src/main/keep-container/keep-content/keep-common-note/keep-common-note.component.ts

```diff
@@ -0,0 +1,162 @@
+        const isArchivedMatch = this.isArchiveNotePresent ? note.isArchived : !note.isArchived;
+        return (this.isSearch && isNotePresent && isArchivedMatch) || (!this.isSearch && isArchivedMatch);
+    }
+
+    selectNoteForEditing(note: Note) {
+        this.selectedNote = note;
+        this.dialogBoxOpen = true;
+        note.showLabelDropdown = false;
+        note.showDropdown = false;
+        this.openDialog();
+    }
+
+    formattedContent(content: string): string {
+        let updateContent: string = this.convertNewlinesToBreaks(content);
+        return this.highlightSearchTerm(updateContent, this.searchValue);
+    }
+
+    highlightSearchTerm(text: string, searchTerm: string): string {
+        if (!searchTerm) {
+            return text;
+        }
+        const regex = new RegExp(searchTerm, 'gi');
+        return text.replace(regex, (match) => {
+            return `<span class="highlight">${match}</span>`;
+        });
+    }
+
+    handleMoreIconClick(event: MouseEvent, note: Note) {
+
+        this.noteService.closeDropdownsExceptThisNote(note);
+        note.showDropdown = !note.showDropdown;
+        if (note.showLabelDropdown) {
+            note.showLabelDropdown = !note.showLabelDropdown;
+            this.searchLabelText = '';
+        }
+        event.stopPropagation();
+    }
+
+    isLabelsExist(note: Note): boolean {
+        return note.labels.length > 0;
+    }
+
+    archiveNote(id: number) {
+        this.notes$ = of(this.noteService.archiveNote(id)).pipe(switchMap(() => {
+            return this.noteService.getNotes();
+        }));
+        this.selectedNote = null;
+    }
+
+    convertNewlinesToBreaks(text: string): string {
+        return text.replace(/\n/g, '<br>');
+    }
+
+    UnArchiveNote(id: number) {
+        this.notes$ = of(this.noteService.unArchiveNote(id)).pipe(switchMap((currentNotes) => {
+            return this.noteService.getNotes();
+        }));
+        this.selectedNote = null;
+    }
+
+    openDialog(): void {
+        this.dialogBoxOpen = true;
+        if (this.selectedNote) {
+            this.dialog.open(KeepCommonEditorComponent, {
+                data: {
+                    Note: this.selectedNote, dialogBoxOpen: this.dialogBoxOpen
+                }
+            });
+
+            this.dialog.afterAllClosed.subscribe(() => {
+                this.dialogBoxOpen = false;
+                this.selectedNote = null;
+            });
+        }
+    }
+
+    @HostListener('document:click', ['$event']) onDocumentClick(event: Event): void {
+
+        const isClickInsideNote = this.notes?.nativeElement?.contains(event.target);
+        if (this.notes && !isClickInsideNote) {
+            this.notes$.forEach((notes) => {
+                notes.forEach((note) => {
+                    note.showDropdown = false;
+                    note.showLabelDropdown = false;
+                });
+            });
+        }
+    }
```

Comment:
do we need this document click?

Reply:
Yes When dropdown menu or label dropdown of any note is open and user click outside the note then they should be closed